### PR TITLE
Feature/support https rpc client

### PR DESCRIPTION
**Pull request by ruseinov**
* [ ] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [ ] Wrote tests
* [x] Updated CHANGELOG.md

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ IMPROVEMENT
 
 - [pubsub] Set default capacity to 0
 - [docs] Various improvements
+- [rpc/client] Supports https and wss now
```
Comment (melekes) [inline CHANGELOG.md]: please move to `0.21.2`. 0.21.0 has been released already

File: rpc/lib/client/http_client.go [hunk 2]
```diff
@@ -36,27 +39,31 @@ func makeHTTPDialer(remoteAddr string) (string, func(string, string) (net.Conn,
...
 		msg := fmt.Sprintf("Invalid addr: %s", remoteAddr)
-		return msg, func(_ string, _ string) (net.Conn, error) {
+		return clientProtocol, msg, func(_ string, _ string) (net.Conn, error) {
 			return nil, errors.New(msg)
 		}
 	}
-	// accept http as an alias for tcp
-	if protocol == "http" {
+	// accept http as an alias for tcp and set the client protocol
+	if protocol == "http" || protocol == "https" {
```
Comment (melekes) [inline rpc/lib/client/http_client.go]: `switch` might be better here

**Comment by melekes** [comment]
Thanks for working on this!

**Comment by ruseinov** [comment]
@melekes np, will fix it shortly.

**Comment by ruseinov** [comment]
all done, feel free to merge

**Comment by melekes** [comment]
linter errors, sorry
```
rpc/lib/client/http_client.go:52:3:warning: redundant break statement (S1023) (gosimple)
rpc/lib/client/http_client.go:55:3:warning: redundant break statement (S1023) (gosimple)
```

File: rpc/lib/client/http_client.go [hunk 3]
```diff
@@ -25,7 +25,10 @@ type HTTPClient interface {
 }
 
 // TODO: Deprecate support for IP:PORT or /path/to/socket
-func makeHTTPDialer(remoteAddr string) (string, func(string, string) (net.Conn, error)) {
+func makeHTTPDialer(remoteAddr string) (string, string, func(string, string) (net.Conn, error)) {
+	// protocol to use for http operations, to support both http and https
+	clientProtocol := "http"
```
Comment (alexanderbez) [inline rpc/lib/client/http_client.go]: Do you feel that `http/s`, `wss`, etc... should be constants in this package?

**Comment by ruseinov** [comment]
@melekes Implemented the comments and checks seem to have passed too.

**Pull request merged**